### PR TITLE
Rename strip reflect to strip nonsemantic

### PR DESCRIPTION
**Pull request by s-perron**
In https://github.com/KhronosGroup/SPIRV-Tools/pull/3110, the strip reflect
pass was changed to also remove all explicitly nonsemantic instructions.  This
makes it so that the name of the pass no longer reflects what the pass actually
does.  This change renames the pass so that it reflects what the pass actaully does.

### CODE REVIEW

File: include/spirv-tools/optimizer.hpp [hunk 1]
```diff
@@ -230,13 +230,14 @@ Optimizer::PassToken CreateNullPass();
 // Section 3.32.2 of the SPIR-V spec) of the SPIR-V module to be optimized.
 Optimizer::PassToken CreateStripDebugInfoPass();
 
-// Creates a strip-reflect-info pass.
-// A strip-reflect-info pass removes all reflections instructions.
-// For now, this is limited to removing decorations defined in
-// SPV_GOOGLE_hlsl_functionality1.  The coverage may expand in
-// the future.
+// [Deprecated] This will not create a strip-nonsemantic-info pass.  See below.
```
Comment (dneto0) [inline include/spirv-tools/optimizer.hpp:233]: Do you mean "This will create a "....  (drop the "not") ?

**Review by dneto0 — CHANGES_REQUESTED** [review]

File: include/spirv-tools/optimizer.hpp (hunk 1 above)
Comment (s-perron) [reply include/spirv-tools/optimizer.hpp]: I do this typo way too often.  I intended to write "now".  However, that is not needed either.  Fixed.

**Review by dneto0 — APPROVED** [review]
Thanks!

**Comment by dneto0** [comment]
Have to update the Android.mk file to react to the renamed .cpp file.

**Review by dneto0 — APPROVED** [review]

**Pull request merged**